feat(Brand): consumed Penta updates

mfrances17 opened the pull request:
<!-- What changes are being made? Please link the issue being addressed. -->
**What**: Closes #9921 

Mostly icon changes, added some extra code to make it responsive in different screen sizes and to make it work with light/dark mode.

<!-- Are there any upstream issues or separate issues you need to reference? -->
**Additional issues**:
None

patternfly-build commented:
Preview: https://patternfly-react-pr-10045.surge.sh

andrew-ronaldson reviewed (approved):
Approved. Great to have the light/dark switcher working on this too!

thatblindgeye commented on packages/react-core/src/components/Brand/examples/BrandBasic.tsx:
```diff
@@ -1,7 +1,15 @@
-  <Brand src={pfLogo} alt="Patternfly Logo" widths={{ default: '200px', md: '400px', xl: '600px' }} />
+  <>
+    <div className="show-light">
+      <Brand src={pfLogo} alt="Patternfly Logo" widths={{ default: '200px', md: '400px', xl: '600px' }} />
+    </div>
+    <div className="show-dark">
+      <Brand src={pfLogoDark} alt="Patternfly Logo" widths={{ default: '200px', md: '400px', xl: '600px' }} />
```
For the alt text in these examples, can we just omit the word "logos", so just `alt="PatternFly"`.

mfrances17 replied:
sounds good, done!

thatblindgeye commented on packages/react-core/src/components/Brand/examples/BrandResponsive.tsx:
```diff
@@ -1,19 +1,36 @@
-    <source media="(min-width: 768px)" srcSet={logoMd} />
-    <source media="(min-width: 576px)" srcSet={logoSm} />
-    <source srcSet={logo} />
-  </Brand>
+  <>
+    <div className="show-light">
+      <Brand src={pfLogo} alt="Fallback patternfly default logo" widths={{ default: '40px', sm: '60px', md: '220px' }}>
```
Similarly we should just pass "PatternFly" as the alt text here.

thatblindgeye replied:
The `show-dark` one below looks good, we just need this one's alt text updated as well

mfrances17 replied:
Strange, I must've done too many undos or something cuz I know it was done at some point... 😄 thanks for catching! Done!

thatblindgeye reviewed (commented):
Couple of quick comments below. @edonehoo do you think it'd be worth adding a description to the basic example to call out that theme support has to be manually done by the consumer, rather than it being something built into the component?

mcoker reviewed (approved):
LGTM 💪

edonehoo commented:
> Couple of quick comments below. @edonehoo do you think it'd be worth adding a description to the basic example to call out that theme support has to be manually done by the consumer, rather than it being something built into the component?

@thatblindgeye I think that makes sense here! wdyt about something simple like this?:

> The brand component does not have any built-in theme support. If a brand should be dynamically styled, you must manually set up the necessary logic on your own. 
> 
> This example adjusts styling as the PatternFly website is toggled between the light and dark theme.

wise-king-sullyman reviewed (approved):
Other than the above comments LGTM!

mfrances17 commented:
> @thatblindgeye I think that makes sense here! wdyt about something simple like this?:
> 
> > The brand component does not have any built-in theme support. If a brand should be dynamically styled, you must manually set up the necessary logic on your own.
> > This example adjusts styling as the PatternFly website is toggled between the light and dark theme.

I added this, slight wording changes and placement... I added the blurb above the basic example under Examples because it applies to both examples.

thatblindgeye reviewed (approved):
🎉

tlabaj reviewed (approved):
LGTM!
